localstack-core/localstack/services/lambda_/provider.py

```diff
@@ -852,6 +853,30 @@ def _validate_layers(self, new_layers: list[str], region: str, account_id: str):
                 )
             visited_layers[layer_arn] = layer_version_arn
 
+    def _validate_capacity_provider(
```

Comment:
nit: `_validate_capacity_provider_config` sounds more accurate and could avoid mixing it up with `localstack.pro.core.services.lambda_.api_utils.validate_capacity_provider` (my initial thought was, don't we have that in pro already)